src/scripts/main.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handleEvent = () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
```

Comment:
In this string template, you are using `element.nodeName` and `element.id` directly. It's a good practice to check if these properties are not null or undefined before using them to avoid possible errors.